From 9c1ab2d3c7c9f0b255a77e61185552ff61202260 Mon Sep 17 00:00:00 2001
From: Nikolaus Rath <Nikolaus@rath.org>
Date: Mon, 1 Feb 2016 08:40:44 -0800
Subject: Force use of libattr1-dev API

Forwarded: not-needed
Patch-Name: force_xattr_api.diff

Compilation under GNU/kFreeBSD fails because none of upstream's
preprocessor macros matches this combination. It is also not clear
how to properly test for the extended attribute API without
introducing something heavyweight like autoconf. In Debian, however,
the libattr1-dev API is available for all supported architectures,
so it is safe to just hardcode the use of this API. This is what
this patch does.
---
 src/llfuse.h | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/src/llfuse.h b/src/llfuse.h
index 159e..1606 100644
--- a/src/llfuse.h
+++ b/src/llfuse.h
@@ -12,18 +12,7 @@ the terms of the GNU LGPL.
 #define PLATFORM_BSD 2
 #define PLATFORM_DARWIN 3
 
-#ifdef __gnu_linux__
 #define PLATFORM PLATFORM_LINUX
-#elif __FreeBSD__
-#define PLATFORM PLATFORM_BSD
-#elif __NetBSD__
-#define PLATFORM PLATFORM_BSD
-#elif __APPLE__ && __MACH__
-#define PLATFORM PLATFORM_DARWIN
-#include "darwin_compat.h"
-#else
-#error "Unable to determine system (Linux/FreeBSD/NetBSD/Darwin)"
-#endif
 
 #include <fuse.h>
 
